### PR TITLE
feat(code-preview): add react code preview

### LINKED ISSUE #298: Add React "code preview" to component documentation page
Generate the code preview for React components into the "code" section of the dev-tools pane.

No syntax highlighting for now, we will do that in a followup task.

### PR DESCRIPTION
**Pull request by janechu**
Closes #298

### CODE REVIEW

File: packages/fast-development-site-react/src/components/site/action-bar.tsx [hunk 1]
```diff
@@ -8,6 +8,11 @@ import { glyphBuildingblocks, glyphExamples, glyphPage } from "@microsoft/fast-g
 import ComponentViewToggle from "./component-view-toggle";
 import manageJss, { ComponentStyles, ICSSRules, IJSSManagerProps, IManagedClasses } from "@microsoft/fast-jss-manager-react";
 
+export enum actionEnum {
```
Comment (chrisdholt) [inline packages/fast-development-site-react/src/components/site/action-bar.tsx]: I think we've been capitalizing enums throughout all projects, should we do so here? Or, is there a principle for not doing so here?

File: packages/fast-development-site-react/src/components/site/dev-tools-code-preview.tsx [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+    private getCodePreview(): string {
+        let codePreview: string = "TBD";
+        this.variables = "";
+
+        switch (this.props.framework) {
+            case frameworkEnum.react:
+                codePreview = this.generateReactCodePreview(this.props.componentName, this.props.data, "", "");
+                break;
+        }
+
```
Comment (nicholasrice) [inline packages/fast-development-site-react/src/components/site/dev-tools-code-preview.tsx]: isn't `this.variables` an empty string here? - is it necessary to concat these?

File: packages/fast-development-site-react/src/components/site/dev-tools-code-preview.tsx [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+    /**
+     * Derive camel cased class name from hyphenated name
+     */
+    private convertToCamelCase: (itemName: string) => string = (itemName: string): string => {
+       itemName = `${itemName.charAt(0).toUpperCase()}${itemName.slice(1)}`;
+
+       return itemName.replace(/-[a-z]/g, (a: string): string => {
+           return a.charAt(1).toUpperCase();
+       });
+    }
```
Comment (chrisdholt) [inline packages/fast-development-site-react/src/components/site/dev-tools-code-preview.tsx]: Can we not use lodash-es to convert to camel case? `.camelCase()`

File: packages/fast-development-site-react/src/components/site/action-bar.tsx (hunk 1 above)
Comment (chrisdholt) [reply packages/fast-development-site-react/src/components/site/action-bar.tsx]: Also seeing this with frameworkEnum

File: packages/fast-development-site-react/src/components/site/dev-tools-code-preview.tsx [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+            for (let i: number = childrenLocations.length - 1; i >= 0; i--) {
+                let stringDataWithReactComponentNames: string = JSON.stringify(propertyChildren[i].data.props, null, 2);
+
+                stringDataWithReactComponentNames = this.replaceReactChildrenVariableIdsWithComponent(
+                    childrenLocations,
+                    propertyChildren,
+                    stringDataWithReactComponentNames
+                );
+
+                this.variables += `var ${propertyName}${propertyChildren[i].location.replace(/children|props|\./g, "")} = `;
```
Comment (nicholasrice) [inline packages/fast-development-site-react/src/components/site/dev-tools-code-preview.tsx]: `const` is pretty universally supported these days - we can probably use it here.

File: packages/fast-development-site-react/src/components/site/dev-tools-code-preview.tsx [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+     */
+    private getReactAttributeObject(location: string, property: string, data: any, tab: string): string {
+        const propertyName: string = location === ""
+            ? property
+            : location + property.charAt(0).toUpperCase() + property.slice(1);
+        const childrenLocations: string[] = this.findChildren(data[property], "", []);
+        const propertyChildren: IReactChildren[] = this.getReactChildrenFromProperties(data, property, propertyName, childrenLocations);
+        const attributes: string = `\n${tab}${property}={${propertyName}}`;
+
+        this.variables += `var ${propertyName} = `;
```
Comment (nicholasrice) [inline packages/fast-development-site-react/src/components/site/dev-tools-code-preview.tsx]: same here with `const`

File: packages/fast-development-site-react/src/components/site/dev-tools-code-preview.tsx [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+            JSONStringWithReactChildComponents = JSONStringWithReactChildComponents.replace(`"${child.id}"`, child.component);
+        }
+
+        return JSONStringWithReactChildComponents;
+    }
+
+    /**
+     * Derive camel cased class name from hyphenated name
+     */
+    private convertToCamelCase: (itemName: string) => string = (itemName: string): string => {
```
Comment (nicholasrice) [inline packages/fast-development-site-react/src/components/site/dev-tools-code-preview.tsx]: lodash has a camelCase function

File: packages/fast-development-site-react/src/components/site/dev-tools-code-preview.tsx (hunk 2 above)
Comment (janechu) [reply packages/fast-development-site-react/src/components/site/dev-tools-code-preview.tsx]: no, this.variables gets added to during the this.generateReactCodePreview()

**Pull request merged**